explore/src/main/scala/explore/config/SpectroscopyModesTable.scala

```diff
@@ -385,26 +389,48 @@ object SpectroscopyModesTable {
             )
         val (enabled, disabled) = rows.partition(enabledRow)
         enabled ++ disabled
-      })
+      }
       // itc results cache
-      .useSerialStateView(
-        // .useState(
+      .useState {
         ItcResultsCache(
-          Map.empty[ITCRequestParams, EitherNec[ItcQueryProblems, ItcResult]]
+          Map.empty[ItcRequestParams, EitherNec[ItcQueryProblems, ItcResult]]
         )
+      }
+      // itcProgress
+      .useState(none[Progress])
```

Comment:
I'm trying to understand how this works with a `worker` now. Before, with a single effect, this component would only get ITC results that it requested, right? What happens now if you submit a bunch of requests on one observation and switch to a new observation which submits a bunch of new requests? Will the new "observation" be processing responses to the requests from the old observation? Or does something prevent that?

Reply:
Good question 🤔

Reply:
Seems like that would throw off the Progress if nothing else.

Reply:
Well progress gets reset if conditions change but I haven't thought about the case where we switch to another page
Maybe we need to index the cache by obsid